plugins/modules/rudder_node_settings.py

```diff
@@ -257,38 +258,36 @@ def _send_request(
             rudder_url=self.rudder_url, path=path
         )
 
-        if method == 'POST':
-            resp = requests.post(
-                url=full_url,
-                headers=self.headers,
-                data=s_data,
-                params=params,
-                verify=verify,
-            )
+        resp, info = fetch_url(
+            method=method,
+            url=full_url,
+            data=s_data,
+            headers=self.headers,
+            verify=verify,
+        )
 
-        elif method == 'GET':
-            resp = requests.get(
-                url=full_url,
-                headers=self.headers,
-                params=params,
```

Comment:
I think that you want to keep the params for GET content. I believe it was used when the query parameter was not empty, a simple data json won't do here.
https://docs.rudder.io/api/v/13/#operation/listAcceptedNodes

Reply:
fixup !